java-frontend/src/main/java/org/sonar/java/resolve/JavaSymbol.java

```diff
@@ -678,12 +681,13 @@ public Boolean checkOverridingParameters(MethodJavaSymbol overridee, ClassJavaTy
         // Generics type should have same erasure see JLS8 8.4.2
 
         JavaType overrideeType = overridee.getParametersTypes().get(i);
-        if (classType.isParameterized()) {
+        overrideeType = resolve.typeSubstitutionSolver.applySiteSubstitutionToFormalParameters(Collections.singletonList(overrideeType), classType).get(0);
+        /*if (classType.isParameterized()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)